### PR TITLE
feat: use retry provider in api

### PR DESCRIPTION
**Pull request by dohaki**
Closes ACX-2582

### CODE REVIEW

File: api/_utils.ts [hunk 1]
```diff
@@ -713,31 +713,41 @@ export const getProvider = (
...
-  return new ethers.providers.StaticJsonRpcProvider(providerUrl);
+  return new sdk.providers.RetryProvider(
+    urls.map((url) => [url, chainId]),
+    chainId,
+    1, // quorum can be 1 in the context of the API
+    3, // retries
+    500, // delay
+    5, // max. concurrency
+    "QUOTES_API", // cache namespace
+    0 // disable RPC calls logging
```
Comment (dohaki) [inline api/_utils.ts:749]: @pxrl @james-a-morris you think these make sense ?

Comment (gsteenkamp89) [reply api/_utils.ts:749]: Why disable logging?

**Review by gsteenkamp89 — APPROVED** [review]

File: api/_utils.ts (hunk 1 above)
Comment (james-a-morris) [reply api/_utils.ts:749]: I think we should disable the logging for the end user. Maybe we can have this be conditioned on whether or not we're running in development/preview mode?

Comment (james-a-morris) [reply api/_utils.ts:749]: These are sensible defaults - we can definitely adjust them as needed

**Review by james-a-morris — APPROVED** [review]

**Pull request merged**